scripts/smb-enum-services.nse

```diff
@@ -0,0 +1,1144 @@
+local msrpc = require "msrpc"
+local smb = require "smb"
+local smbauth = require "smbauth"
+local stdnse = require "stdnse"
+local string = require "string"
+local table = require "table"
+local http = require "http"
+local shortport = require "shortport"
+
+description = [[
+Retrieves the list of services running on a remote Windows system.
+Each service attribute contains service name, display name and service status of
+each service.
+
+References:
+* https://msdn.microsoft.com/en-us/library/windows/desktop/ms682637(v=vs.85).aspx
+* https://msdn.microsoft.com/en-us/library/windows/desktop/ms682651(v=vs.85).aspx
+* https://github.com/samba-team/samba/blob/d8a5565ae647352d11d622bd4e73ff4568678a7c/librpc/idl/svcctl.idl
+]]
+
+-- @usage
+-- nmap --script smb-enum-services.nse -p445 <host>
+-- nmap --script smb-enum-services.nse --script-args smbusername=<username>,smbpass=<password> -p445 <host>
+--
+-- The following lines displays the normal and xml results when this script
+-- was run against Windows 2003 R2 x64 Enterprise Server.
+--
+-- @output
+-- | smb-enum-services:
+-- |
+-- |     displayName: Application Layer Gateway Service
+-- |     serviceName: ALG
+-- |     serviceStatus:
+-- |       check_point: 0
+-- |       wait_hint: 0
+-- |       state:
+-- |
```

Comment:
`state` isn't showing up, which is because it's being mixed up with a different enum. The constants in `msrpctypes.lua` for `svcctl_State` are for the `dwServiceState` **input** parameter, and are unrelated to the actual values [defined for `dwCurrentState` in the SERVICE_STATE struct](https://msdn.microsoft.com/en-us/library/windows/desktop/ms685996(v=vs.85).aspx). I don't see that these are used anywhere else, either, so just change the contents of the `svcctl_State` table in `msrpctypes.lua` to match the values in that reference, which ought to fix it.

Reply:
Should I change the table being passed in here, https://github.com/nmap/nmap/blob/master/nselib/msrpctypes.lua#L4470 ? Or should I create a duplicate of msrpctypes.unmarshall_SERVICE_STATUS function and change this line, https://github.com/nmap/nmap/blob/master/nselib/msrpctypes.lua#L4515 ?